### PR TITLE
RDM-3272

### PR DESCRIPTION
**Pull request by tzarsmango**
Create a new CCD application gateway resource to divert all requests to `/documents` path to Palo Alto for AV inspection. All other requests will go through the ccd-api-gateway as usual. The documents containing any virus would be rejected by the Palo Alto module.

### CODE REVIEW

File: demo.tfvars [hunk 1]
```diff
@@ -0,0 +1,4 @@
+external_hostname_gateway = "gateway-ccd.demo.platform.hmcts.net"
+external_hostname_www = "www-ccd.demo.platform.hmcts.net"
+external_cert_name = "star-demo"
```
Comment (dwaynebailey) [inline demo.tfvars]: Is that cert name correct?

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -11,11 +11,20 @@ slackChannel = "#ccd-master-builds"
...
-        sharedInfrastructurePipeline('ccd', environment.demoName, 'nonprod')
         sharedInfrastructurePipeline('ccd', environment.prodName, 'prod')
         notifyBuildFixed channel: slackChannel
     } catch (err) {
         notifyBuildFailure channel: slackChannel
         throw err
     }
 }
+
+onDemo {
```
Comment (dwaynebailey) [inline Jenkinsfile_CNP]: Do we want to keep this demo deploy on its own?  Maybe for this initial PR but we should remove in final

File: state.tf [hunk 3]
```diff
@@ -1,3 +1,9 @@
 terraform {
   backend "azurerm" {}
 }
+
+data "azurerm_subnet" "main_subnet" {
```
Comment (dwaynebailey) [inline state.tf]: Rather move this to where it will be used i.e. in the GW config

Comment (dwaynebailey) [inline state.tf]: IIRC this is the subnet for the ASE? Can we name it as such as main_subnet is a bit confusing.

**Comment by tzarsmango** [comment]
Resolved by https://github.com/hmcts/ccd-shared-infrastructure/pull/29. Closing and deleting...

**Pull request closed without merging**